myia/parser.py

```diff
@@ -42,9 +42,31 @@
 import inspect
 import textwrap
 from types import FunctionType
-from typing import Any, Dict, List, Optional, Tuple, Type
+from typing import overload, Any, Dict, List, Optional, Tuple, Type
 
 from myia.anf_ir import ANFNode, Parameter, Apply, Graph, Constant
+from .info import DebugInfo, about
+
+
+class Location:
+    """A location in source code.
+
+    Attributes:
+        url: The filename.
+        line: The line number.
+        column: The column number.
+
+    """
+
+    def __init__(self, url, line, column):
+        """Initialize a Location."""
+        self.url = url
+        self.line = line
+        self.column = column
```

Comment:
Since this is basically just a struct, maybe a good idea to make it immutable (and leave the pretty string printing to e.g. the logger).
```python
class Location(NamedTuple):
    url: str
    line: str
    column: int
```

Reply:
That's a good idea.